src/ui/components/NavigationBar.js

```diff
@@ -0,0 +1,94 @@
+import { testStep } from "../../common/helpers/pwHelpers.js";
+
+export class NavigationBar {
+  constructor(page, userId = 0) {
+    this.page = page;
+    this.userId = userId;
+  }
+
+  step(title, fn) {
+    return testStep(title, fn, this.userId);
+  }
+
+  async clickNavLink(name) {
+    await this.page.getByRole("link", { name }).click();
+  }
+
+  async waitForPageUrl(urlPattern) {
+    await this.page.waitForURL(urlPattern, {
+      waitUntil: "domcontentloaded",
+    });
+  }
+
+  async waitForVisible(selector) {
+    await this.page.locator(selector).waitFor({ state: "visible" });
+  }
+
+  async waitForLoggedOut() {
+    await this.page.waitForURL(/\/(index|login)\.htm/i, {
+      waitUntil: "domcontentloaded",
+      timeout: 10000,
+    });
+
+    await this.page
+      .getByRole("button", { name: "Log In" })
+      .waitFor({ state: "visible", timeout: 5000 });
+  }
+
+  go = {
```

Comment:
this `go` is just redundant and you can use all of these methods without it

and if you asserting what page is opened you should mention it in the methods names